### PR TITLE
Implement generic OTP action.

### LINKED ISSUE #422: Moving to an MIT licensed OTP library?
I am looking at unifying the OTP interface. Having to specify TOTP or HOTP is no fun. A better alternative is to have a generic `otp` action that works with either. This is how [browserpass](https://github.com/dannyvankooten/browserpass) does it.

I haven't started writing any of the code because I wanted to get opinions on switching to [gokyle/twofactor](https://github.com/gokyle/twofactor). My reasons for wanting to switch are:
- MIT licensed 
- Fewer 3rd party dependencies (only one: https://godoc.org/rsc.io/qr)
- Simpler API that has one entry point for HOTP and TOTP.

### PR DESCRIPTION
**Pull request by qbit**
- support for TOTP and HOTP urls.
- use 'totp' as an alias for new otp action.
- add ability to parse "otpauth://" urls (supported by pass-otp,
  passforios, browserpass).
- maintain compat with YAML style "totp" entries.
- add ability to produce QR codes (-qr filename.png).

Closes #422

### CODE REVIEW

File: action/otp.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	}
+
+	sec, err := s.Store.Get(ctx, name)
+	if err != nil {
+		return s.exitError(ctx, ExitDecrypt, err, "failed to get entry '%s': %s", name, err)
+	}
+
+	otpURL := ""
+	for _, line := range strings.Split(sec.Body(), "\n") {
+		matched, err := regexp.MatchString("^otpauth:", line)
```
Comment (dominikschulz) [inline action/otp.go]: I'd prefer `strings.HasPrefix`, is there any reason to use an regexp here?

File: main.go [hunk 2]
```diff
@@ -582,20 +582,46 @@ func main() {
 		},
 		{
```
Comment (dominikschulz) [inline main.go]: We should remove this subcommand and just set `Aliases: []string{"totp", "htop"}` on the `otp` subcommand.

**Comment by qbit** [comment]
Updated!

**Comment by pakohan** [comment]
I'm facing this issue now, do I need to call the old command differently now or is it broken for all my old entries and I need to edit them?

```
$ gopass  gitlab.blablabla.com/authenticator
otpauth://totp/john.doe@blablabla.com?secret=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX&issuer=gitlab.blablabla.com+%7C+john.doe%40blablabla.com
$ gopass totp -c gitlab.blablabla.com/authenticator

Error: No OTP entry found for gitlab.blablabla.com/authenticator
$```

File: docs/features.md [hunk 3]
```diff
@@ -132,6 +132,33 @@ $ gopass edit golang.org/gopher
...
+
+*Note: Depending on your security needs, it may not behoove you to store your OTP secrets alongside your passwords! Look into [Multiple Stores](https://github.com/justwatchcom/gopass/blob/master/docs/features.md#multiple-stores) if you need things to be separate!*
+
+Typically sites will display a QR code containing a URL that starts with `oauth://`. This string contains information about generating your OTPs and can be directly added to your password file. For example:
+
+```
+gopass golang.org/gopher
+secret1234
+otpauth://totp/golang.org:gopher?secret=ABC123
+```
```
Comment (arvenil) [inline docs/features.md]: @dominikschulz I don't get those instructions or this is really faulty.
With both secret/password and totp in one entry getting a password is a mess. It doesn't get copied automatically anymore.

Without totp works correctly, copies password to clipboard
```term
$ gopass show golang.org/gopher
No safe content to display, you can force display with show -f.
Copying password instead.
✔ Copied golang.org/gopher to clipboard. Will clear in 45 seconds.
```
```
$ gopass show -f golang.org/gopher
secret1234
```
```
$ gopass totp golang.org/gopher

Error: No OTP entry found for golang.org/gopher: illegal base32 data at input byte 0
```
Edit and add totp line.
```
$ gopass edit golang.org/gopher
```
totp works nicely
```
gopass totp golang.org/gopher
493858 lasts 27s 	|---===========================|
```
show -f shows everything, ok fine
```
$ gopass show -f golang.org/gopher
secret1234
otpauth://totp/golang.org:gopher?secret=ABC123
```
**Doesn't copy anymore to clippboard**, but properly doesn't show password
```
$ gopass golang.org/gopher
otpauth://totp/golang.org:gopher?secret=ABC123
```
I need to use flag again to copy:
```
$ gopass show -c golang.org/gopher
✔ Copied golang.org/gopher to clipboard. Will clear in 45 seconds.
```

Comment (dominikschulz) [reply docs/features.md]: That looks confusing indeed, however it's also confusing to discuss this in an old PR. Could you copy your comment to a new issue referencing this PR, so we can properly discuss it and see how we can improve the UX?

**Pull request merged**